src/modules/tests/tests.service.ts

```diff
@@ -420,46 +420,123 @@ export class TestsService {
     }
 
     // Get all attempts for this user and test
-    const attempts = await this.testRepository.manager.find(TestAttempt, {
+    const allAttempts = await this.testRepository.manager.find(TestAttempt, {
       where: {
         testId,
         userId,
         tenantId: authContext.tenantId,
         organisationId: authContext.organisationId,
       },
-      order: { startedAt: 'DESC' },
+      order: { startedAt: 'ASC' },
     });
 
-    const totalAttempts = attempts.length;
+    const totalAttempts = allAttempts.length;
     const maxAttempts = test.attempts;
 
-    if (attempts.length === 0) {
-      // No attempts yet - user can start a new attempt
-      return {
-        canResume: false,
-        canReattempt: true,
-        lastAttemptStatus: null,
-        lastAttemptId: null,
-        maxAttempts,
-        totalAttempts,
-      };
+    // Check if there are any attempts under review
+    const hasPendingReview = allAttempts.some(attempt => 
```

Comment:
@vinayak-patil if totalAttempts=0 wont this give error?

Reply:
To be on saferside, if the test isObjective is true, there wont be any review so you should not have conditions for review pending or submitted